Tools/module_config/generate_actuators_metadata.py

```diff
@@ -264,7 +264,27 @@ def get_config_params(param_list):
             }
         per_channel_params.append(param)
 
-        # TODO: support non-standard per-channel parameters
+        # custom-per-channel-params
+        custom_params = group.get('custom_params', {})
```

Comment:
Can you follow the existing structure?
Instead of:
```
        pwm:
          label: PWM Mode
          prefix: PWM
          description: |
            Enable PWM mode will make PCA9685 output duty cycle instead of PPM pulse.
            Using MIN value different from 0 will lead to output (MIN/4096)% instead of 0%.
            Same applies to MAX and other fields.

          advanced: true
          param:
            type: boolean
            default: false
```
It should be a list of parameters:
```
              - name: 'PCA9685${i}_MODE'
                label: 'PWM Mode'
                index_offset: 0
                advanced: false
```
And extend the schema: https://github.com/PX4/PX4-Autopilot/blob/main/validation/module_schema.yaml#L285

Then use the existing parameters section (https://github.com/PX4/PX4-Autopilot/blob/main/validation/module_schema.yaml#L83) to define the parameter:
`module.yml`:
```
parameters:
    - group: PWM
      definitions:
        PCA9685${i}_MODE:
            description:
                short: ...
                long: |
                    Enable PWM mode will make PCA9685 output duty cycle instead of PPM pulse.
            Using MIN value different from 0 will lead to output (MIN/4096)% instead of 0%.
            Same applies to MAX and other fields.
            type: boolean
            default: false
            num_instances: 16
```

-----------
With that, bitfield params are supported as well: you would declare a single parameter, and then in the output section use:
```
              - name: 'PCA9685_MODE'
                label: 'Enable PWM Mode'
                show_as: bitset
```

Reply:
Ohh thanks for pointing out the correct way! I've never figured out how those params are defined and validated this before. Will update it soon :)